src/util-thash.c

```diff
@@ -311,7 +311,8 @@ THashTableContext *THashInit(const char *cnf_prefix, size_t data_size,
     THashTableContext *ctx = SCCalloc(1, sizeof(*ctx));
     BUG_ON(!ctx);
 
-    ctx->config.data_size = data_size;
+    BUG_ON(data_size > UINT32_MAX);
```

Comment:
will this give new warnings on 32bit? There `size_t` is 32bit so `> UINT32_t` should be invalid

Reply:
Lets change `data_size`'s type to `uint32_t`. I don't like these implicit types like `size_t` in our own API